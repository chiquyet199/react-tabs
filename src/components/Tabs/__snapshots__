/Users/edward/Desktop/canopy-frontend/packages/canopy-frontend-common/src/components/Tabs/__native_snapshots__