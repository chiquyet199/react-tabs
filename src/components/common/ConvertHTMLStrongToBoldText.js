import React from "react";
import {Text} from "react-native";
import styled from "styled-components/native";

export const StyledView = styled(Text)`
  display: inline;
  flex-direction: row;
  flex-wrap: wrap;
  align-items: center;
  justify-content: center;
`;

export default ({htmlString}) => {
  const parsed = [];
  let cursor = 0;
  let searchingFor = "<strong>";
  while (cursor < htmlString.length) {
    let endPoint = htmlString.indexOf(searchingFor, cursor);
    endPoint = endPoint === -1 ? htmlString.length : endPoint;
    const token = htmlString.substring(cursor, endPoint);
    if (searchingFor === "<strong>") {
      parsed.push(<Text key={token}>{token}</Text>);
    } else {
      // Styled-components seems to ignore font-weight here
      parsed.push(
        <Text style={{fontWeight: "bold"}} key={token}>
          {token}
        </Text>,
      );
    }
    cursor = endPoint + searchingFor.length;
    searchingFor = searchingFor === "<strong>" ? "</strong>" : "<strong>";
  }

  return <StyledView>{parsed}</StyledView>;
};
