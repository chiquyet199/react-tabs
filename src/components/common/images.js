import styled from "styled-components/native";

export const CenterIconImage = styled.Image`
  width: 80px;
  height: 80px;
  margin-top: 20px;
  margin-bottom: 20px;
  align-self: center;
`;

export const CenterPageImage = styled.Image`
  width: 370px;
  height: 320px;
  margin-top: 20px;
`;
