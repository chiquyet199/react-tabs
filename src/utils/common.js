import {Dimensions, Platform} from "react-native";
import React from "react";
import {connect} from "react-redux";
import styled from "styled-components/native";
import styledWeb, {withTheme} from "styled-components";

export const isWeb = Platform.OS === "web";
export const isNative = Platform.OS === "ios" || Platform.OS === "android";
export const isIos = Platform.OS === "ios";
export const isDev = process.env.NODE_ENV !== "production";
export const extractValueFromEvent = event =>
  event.target ? event.target.value : event;
export const extractKeyFromEvent = event =>
  event.nativeEvent ? event.nativeEvent.key : event.key;

export const sleep = ms => new Promise(resolve => setTimeout(resolve, ms));

export const cleanProps = props => {
  // List of attributes to skip.
  const BLACKLISTED_PROPS = ["dispatch"];
  // Initialize the `cleanProps` as an empty object.
  const ownCleanProps = {};
  // Iterate over all the properties (Eslint - loops should be avoided in favor of array iterations)
  Object.keys(props).forEach(propKey => {
    // Check if the key is not in the blacklisted array.
    if (BLACKLISTED_PROPS.indexOf(propKey) === -1) {
      // Append the attribute to the clean object.
      ownCleanProps[propKey] = props[propKey];
    }
  });

  // Return the cleaned properties.
  return ownCleanProps;
};

export const responsive = Component => styles => {
  const ResponsiveComponent = styled(Component)`
    ${({size}) => styles[size]};
  `;

  const ResponsiveWrapper = props => {
    const ownProps = cleanProps(props);

    return <ResponsiveComponent {...ownProps} />;
  };

  return connect(state => ({
    size: state.dimensions.size,
  }))(ResponsiveWrapper);
};

export const responsiveWeb = Component => styles => {
  const ResponsiveWebComponent = styledWeb(Component)`
    ${({size}) => styles[size]};
  `;

  const ResponsiveWebWrapper = props => {
    const ownProps = cleanProps(props);

    return <ResponsiveWebComponent {...ownProps} />;
  };

  return connect(state => ({
    size: state.dimensions.size,
  }))(ResponsiveWebWrapper);
};

export const isX = (() => {
  const d = Dimensions.get("window");
  const {height, width} = d;

  return (
    // This has to be iOS duh
    Platform.OS === "ios" &&
    // Accounting for the height in either orientation
    (height === 812 || width === 812)
  );
})();

// Use theme fallback because Selva has another reducer
export const themed = Component =>
  connect(({globaltheme, theme}) => ({
    globaltheme: globaltheme || theme,
  }))(withTheme(Component));

export const redirectToBaseRoute = (
  goTo,
  form,
  location = {},
  prefixOnly = false,
) => {
  const {pathname} = location;
  const urlArr = pathname.split("/").filter(i => i !== "");
  const [baseUrl] = urlArr;
  if (prefixOnly) {
    return baseUrl;
  }
  const {DynamicForm} = form;
  if (!DynamicForm) {
    goTo(`/${baseUrl}`);
  }

  return true;
};

export const parseStyle = style => {
  return Object.keys(style).reduce((previousValue, currentValue) => {
    const starter = previousValue ? `${previousValue};\n` : "";

    return `${starter}${currentValue}: ${style[currentValue]}`;
  }, "");
};

export const applyStyleObjectToStyledComponent = (style, component) => {
  if (!style || Object.keys(style).length === 0) {
    return component;
  }

  const parsedStyles = parseStyle(style);

  return styled(component)`
    ${parsedStyles};
  `;
};
