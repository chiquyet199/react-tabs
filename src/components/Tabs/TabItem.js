import React from "react";
import styled from "styled-components/native";
import {TabBar} from "react-native-tab-view";
import {themed} from "../../utils/common";

const StyledLabel = themed(styled.Text`
  margin: 8px;
  color: ${({theme, globaltheme, active}) =>
    active
      ? theme[globaltheme].tabs.activeText
      : theme[globaltheme].tabs.inactiveText};
  font-weight: bold;
`);

const StyledTabs = styled(TabBar)`
  background-color: ${({theme, globaltheme}) =>
    theme[globaltheme].tabs.background};
`;

export const TabItem = ({active, theme, globaltheme, ...props}) => {
  return (
    <StyledTabs
      theme={theme}
      globaltheme={globaltheme}
      {...props}
      indicatorStyle={{
        backgroundColor: theme[globaltheme].tabs.indicator,
      }}
      renderLabel={({route}) => (
        <StyledLabel
          active={
            active ||
            props.navigationState.routes[props.navigationState.index].title ===
              route.title
          }
        >
          {route.title}
        </StyledLabel>
      )}
    />
  );
};

export default themed(TabItem);
