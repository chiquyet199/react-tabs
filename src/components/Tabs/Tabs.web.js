import React, {Component} from "react";
import styled from "styled-components/native";
import {array} from "prop-types";
import {themed} from "../../utils/common";

const Container = styled.View`
  flex: 1;
`;

const TabsContainer = styled.View`
  flex-direction: row;
`;

export const TabContainer = themed(styled.TouchableOpacity`
  flex: 1;
  padding-vertical: 15px;
  border-bottom-width: 1px;
  border-bottom-color: ${({active, globaltheme, theme}) =>
    active
      ? theme[globaltheme].tabs.borderBottom
      : theme.colors.semiTransparent};
`);

const TabText = themed(styled.Text`
  font-weight: bold;
  text-align: center;
  opacity: 1;
  color: ${({active, globaltheme, theme}) =>
    active
      ? theme[globaltheme].tabs.activeText
      : theme[globaltheme].tabs.inactiveText};
  text-transform: none;
`);

export class WebTabs extends Component {
  state = {
    index: this.props.index || 0,
  };

  handleTabChange = (index, onClick) => {
    if (onClick) {
      if (onClick() !== false) {
        this.setState({index});
      }
    } else {
      this.setState({index});
    }
  };

  render() {
    const {children} = this.props;

    if (children === undefined || children.length <= 1) {
      return null;
    }

    return (
      <Container>
        <TabsContainer type="TabsContainer">
          {children.map(({props: {title, onClick}}, index) => (
            <TabContainer
              active={index === this.state.index}
              onPress={() => this.handleTabChange(index, onClick)}
              key={title}
            >
              <TabText active={index === this.state.index}>{title}</TabText>
            </TabContainer>
          ))}
        </TabsContainer>
        <Container type="ContentContainer">
          {children[this.state.index]}
        </Container>
      </Container>
    );
  }
}

WebTabs.propTypes = {
  children: array,
};

export default WebTabs;
