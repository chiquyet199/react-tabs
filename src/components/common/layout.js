import React from "react";
import styled from "styled-components/native";
import {isNative, isWeb, responsive, themed} from "../../utils/common";
import {TouchableOpacity} from "react-native";
import {DESKTOP_CONTAINER_WIDTH} from "../../constants/sizes";
import {ButtonContainer} from "../Button/Button";
import {func, node, number, string} from "prop-types";

export const FlexContainer = styled.View`
  flex: 1;
  width: 100%;
  height: 100%;
  max-width: ${({full}) => (full ? "100%" : `${DESKTOP_CONTAINER_WIDTH}px`)};
`;

export const Flexed = styled.View``;

export const Flex = styled.View`
  margin-top: ${({theme}) => theme.padding.top + "px"};
  position: absolute;
  width: 100%;
  background-color: transparent;
`;

export const FlexColumn = styled.View`
  max-width: 537px;
  display: flex;
  flex-direction: column;
  justify-content: center;
`;

export const TopBarContainer = styled.View`
  width: 100%;
  max-width: 100%;
  padding: 0;
  padding-top: ${({theme}) => theme.padding.top + "px"}
  z-index: 2;
  position: relative;
  background-color: ${({theme, globaltheme}) =>
    theme[globaltheme].topbar.background}
`;

export const BaseContainer = styled.View`
  flex: 1;
  padding: 0;
  width: 100%;
  z-index: 1;
`;

export const CenteredContainer = styled.View`
  flex: 1;
  justify-content: ${({align}) => align || "center"};
  align-self: center;
  align-items: center;
  width: 100%;
  max-width: ${({full}) => (full ? "100%" : `${DESKTOP_CONTAINER_WIDTH}px`)};
  margin-horizontal: auto;
  padding-horizontal: ${({full}) => (full === "true" ? "0" : "10px")};
  flex-basis: auto;
`;

export const EndPageContainer = responsive(CenteredContainer)({
  S: `justify-content: flex-start`,
});

export const PageContainer = responsive(styled(CenteredContainer)`
  width: 100%;
  max-width: ${({full}) =>
    full === "true" ? "100%" : `${DESKTOP_CONTAINER_WIDTH}px`};
  align-self: center;
  align-items: stretch;
  margin-top: ${({layout}) => (layout === "hq" ? "60px" : "40px")};
  ${({margintop}) => margintop >= 0 && `margin-top: ${margintop}px`};
  padding-horizontal: 10px;
  justify-content: center;
  flex-basis: auto;
`)({
  S: `
      padding-top: ${({theme}) => theme.padding.top + "px"};
      justify-content: space-between;
      padding-horizontal: 10px;
      margin-top: ${({layout}) => (layout === "hq" ? "80px" : "20px")};
    `,
});

export const FormContainer = responsive(styled.View`
  justify-content: center;
  align-items: center;
  max-width: ${DESKTOP_CONTAINER_WIDTH}px;
  width: 100%;
  align-self: center;
`)({
  S: `
      padding-top: 10px;
      justify-content: space-between;
      max-width: 100%;
    `,
});

export const FlexRow = styled.View`
  flex: 1;
  flex-direction: row;
  flex-basis: 100%;
`;

export const ButtonBar = styled.View`
  max-width: ${({isMenu}) =>
    isMenu ? "100%" : `${DESKTOP_CONTAINER_WIDTH}px`};
  width: 100%;
  border-top-color: ${({isMenu, theme}) =>
    isMenu ? theme.colors.lightSteel : theme.colors.none};
  border-top-width: ${({isMenu}) => (isMenu ? "1px" : 0)};
  background-color: ${({isMenu, theme}) =>
    isMenu ? theme.colors.white : theme.colors.none};
  flex-direction: row;
  margin-horizontal: 0;
  margin-bottom: ${({isMenu}) => (isMenu ? 0 : "5px")};
  flex-wrap: ${({wrap}) => (wrap ? "wrap" : "nowrap")};
  ${({mt}) => mt && `margin-top: ${mt}px}`};
  ${ButtonContainer} {
    margin: 20px;
  }
`;

export const BottomButtonBar = responsive(ButtonBar)({
  S: `flexGrow: 10;
      flexDirection: row;
      alignItems: center;
      justifyContent: flex-end;
    `,
});

export const ResponsiveWebScrollView = responsive(styled.ScrollView``)({
  M: `
        width: 500px;
  `,
  L: `
        width: 600px;
    `,
});

export const PageTitle = themed(styled.Text`
  width: 100%;
  font-size: 28px;
  line-height: 36px;
  color: ${({globaltheme, theme}) => theme[globaltheme].baseColor};
  text-align: ${({left}) => (left ? "left" : "center")};
  padding-horizontal: 5px;
  align-self: ${({left}) => (left ? "flex-start" : "center")};
  font-weight: ${({bold}) => (bold ? "bold" : "normal")};
  ${({mb}) => mb && `margin-bottom: ${mb}px}`};
  ${({color}) => color && `color: ${color}`};
`);

export const TextParagraph = themed(styled.Text`
  width: 100%;
  font-size: 16px;
  line-height: 24px;
  color: ${({disabled, theme, globaltheme}) =>
    disabled
      ? theme[globaltheme].disabledTextColor
      : theme[globaltheme].baseColor};
  font-weight: ${({bold}) => (bold ? "bold" : "normal")};
  font-style: ${({fontStyle}) => fontStyle || "normal"};
  align-self: ${({left}) => (left ? "flex-start" : "center")};
  text-align: ${({left}) => (left ? "left" : "center")};
  padding: ${({gutters}) => gutters || "15px 5px"};
  flex-shrink: 0;
  flex-grow: 0;
  ${({mb}) => mb && `margin-bottom: ${mb}px`};
  ${({color}) => color && `color: ${color}`};
`);

export const ViewParagraph = themed(styled.View`
  width: 100%;
  flex-shrink: 0;
  flex-grow: 0;
`);

export const SubTitle = themed(styled(TextParagraph)`
  font-weight: bold;
  color: ${({globaltheme, theme}) => theme[globaltheme].baseColor};
  ${({color}) => color && `color: ${color}`};
`);

export const TextLinkText = themed(styled.Text`
  text-align: center;
  font-size: 16;
  line-height: 24;
  color: ${({theme, globaltheme}) => theme[globaltheme].accentColor};
  font-weight: 700;
  padding-top: ${({gutters}) => (gutters >= 0 ? `${gutters}px` : "15px")};
  padding-bottom: ${({gutters}) => (gutters >= 0 ? `${gutters}px` : "15px")}
  padding-left: ${({gutters}) => (gutters >= 0 ? `${gutters}px` : "15px")}
  padding-right: ${({gutters}) => (gutters >= 0 ? `${gutters}px` : "15px")}
  text-align:  ${({align}) => align || "center"};
`);

export const TextLink = ({onClick, children, gutters, align, globaltheme}) => (
  <TouchableOpacity onPress={onClick}>
    <TextLinkText align={align} gutters={gutters} globaltheme={globaltheme}>
      {children}
    </TextLinkText>
  </TouchableOpacity>
);

TextLink.propTypes = {
  onClick: func,
  children: node,
  gutters: number,
  align: string,
  globaltheme: string,
};

export const NotchSafety = styled.View`
  height: 40px;
`;

export const PageHeader = styled.Text`
  font-size: 18;
  text-align: center;
  width: 100%;
  margin-top: ${({theme}) => theme.padding.top + 15 + "px"};
  color: ${({globaltheme, theme}) => theme[globaltheme].baseColor};
  position: absolute;
`;

export const HeadlineText = responsive(styled.Text`
  width: 100%;
  font-size: 28px;
  line-height: 36px;
  color: ${({globaltheme, theme}) => theme[globaltheme].baseColor};
  align-self: ${({left}) => (left ? "flex-start" : "center")};
  text-align: ${({left}) => (left ? "left" : "center")};
  margin: 8px 0 40px;
  padding: 0;
`)({
  S: `
    padding-left: 15px;
    `,
  L: `
    align-self: center;
   `,
});

export const BoxButtonBar = styled.View`
  align-self: center;
  align-items: stretch;
  width: 100%;
  margin-horizontal: auto;
  flex-direction: row;
`;

export const FixedWidthContainer = styled.View`
  width: ${DESKTOP_CONTAINER_WIDTH}px;
  margin-horizontal: auto;
  padding-horizontal: 15px;
`;

export const PaddedContent = styled.View`
  padding-horizontal: 10px;
`;

export const ErrorMessage = themed(styled(TextParagraph)`
  color: ${({theme}) => theme.colors.alertRed};
`);

// Scroll layout

export const ScrollContentContainer = styled.View`
  width: 100%;
  ${isNative ? "flex: 1;" : "height: calc(100vh - 52px)"};
  flex-grow: ${() => (isWeb ? 0 : 1)};
  flex-shrink: ${() => (isWeb ? 0 : 1)};
  justify-content: center;
  align-self: center;
  align-items: center;
`;

export const ScrollButtonBar = styled(ButtonBar)`
  padding-horizontal: 10px;
  ${({mt}) => mt && `margin-top: ${mt}px}`};
`;

export const ScrollScrollView = styled.ScrollView`
  ${isNative ? "" : "height: 100%"};
  width: 100%;
  flex-grow: 1;
  flex-shrink: 1;
`;
