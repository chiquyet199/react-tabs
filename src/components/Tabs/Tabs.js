import React, {Component} from "react";
import {Dimensions} from "react-native";
import {TabView, SceneMap} from "react-native-tab-view";
import {array, object} from "prop-types";
import {themed} from "../../utils/common";
import {TabItem} from "./TabItem";

export class Tabs extends Component {
  state = {
    index: this.props.index || 0,
    scene: SceneMap(this.props.scene),
    routes: this.props.routes,
  };

  static getDerivedStateFromProps(props, state) {
    if (props.scene !== state.scene) {
      return {
        scene: SceneMap(props.scene),
      };
    }

    return null;
  }

  render() {
    const {theme, globaltheme} = this.props;
    const {index, routes, scene} = this.state;

    return (
      <TabView
        navigationState={{index, routes}}
        renderScene={scene}
        onIndexChange={idx => this.setState({index: idx})}
        initialLayout={{width: Dimensions.get("window").width, height: 47}}
        renderTabBar={props => TabItem({theme, globaltheme, ...props})}
      />
    );
  }
}

Tabs.propTypes = {
  scene: object.isRequired,
  routes: array.isRequired,
};

export default themed(Tabs);
