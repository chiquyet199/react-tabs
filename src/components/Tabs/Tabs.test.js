import React from "react";
import {View, Text, Dimensions} from "react-native";
import {shallow} from "enzyme";
import {extend} from "underscore";
import {isWeb} from "../../utils/common";
import {Tabs} from "./Tabs";
import {WebTabs, TabContainer} from "./Tabs.web";
import {TabItem} from "./TabItem";

const page1 = () => (
  <View>
    <Text>page1</Text>
  </View>
);
const page2 = () => (
  <View>
    <Text>page1</Text>
  </View>
);

const mockTheme = {
  light: {
    background: ["#ffffff", "#ffffff"],
    accentColor: "#5ab88e",
  },
};

jest.mock("./TabItem");

const initialLayout = {width: Dimensions.get("window").width, height: 47};

describe("Tabs - page tests", () => {
  it("matches snapshot", () => {
    const expectedComponent = isWeb ? getWebComponent() : getNativeComponent();
    expect(expectedComponent).toMatchSnapshot();
  });

  if (isWeb) {
    it("If web tabs zero children", () => {
      const nullChild = getWebComponentZeroChild();
      expect(nullChild.props().children).toBeUndefined();
    });

    it("Check if correct content page", () => {
      const defTabs = getWebComponent();
      const content = defTabs
        .dive()
        .find(View)
        .findWhere(tab => tab.props().type === "ContentContainer");
      const contentView = content.find(View).find(Text);
      expect(contentView.props().children).toEqual("Page 1");
    });

    it("Check if correct default tab active", () => {
      const defTabs = getWebComponent();
      const tabs = defTabs
        .dive()
        .find(View)
        .findWhere(tab => tab.props().type === "TabsContainer");

      const defTab = tabs.find(TabContainer);

      defTab.forEach((view, index) => {
        if (index === 0) {
          expect(view.prop("active")).toBe(true);
        } else {
          expect(view.prop("active")).toBe(false);
        }
      });
    });

    it("Check if correct changes content page", () => {
      const defTabs = getWebComponent();
      const tabs = defTabs
        .dive()
        .find(View)
        .findWhere(tab => tab.props().type === "TabsContainer");
      const defTab = tabs.find(TabContainer);

      defTab.forEach((view, index) => {
        if (index === 1) {
          view.simulate("press");
        }
      });

      defTabs.update();

      const newContent = defTabs
        .dive()
        .find(View)
        .findWhere(tab => tab.props().type === "ContentContainer");

      const newContentView = newContent.find(View).find(Text);

      expect(newContentView.props().children).toEqual("Page 2");
    });

    it("Check if correct switch active tab", () => {
      const defTabs = getWebComponent();
      const tabs = defTabs
        .dive()
        .find(View)
        .findWhere(tab => tab.props().type === "TabsContainer");

      const defTab = tabs.find(TabContainer);

      defTab.forEach((view, index) => {
        if (index === 1) {
          view.simulate("press");
        }
      });

      defTabs.update();

      const updatedTabs = defTabs
        .dive()
        .find(View)
        .findWhere(tab => tab.props().type === "TabsContainer");

      const newDefTab = updatedTabs.find(TabContainer);

      newDefTab.forEach((view, index) => {
        if (index === 0) {
          expect(view.prop("active")).toBe(false);
        } else {
          expect(view.prop("active")).toBe(true);
        }
      });
    });
  } else {
    it("Check native tab onIndexChange ", () => {
      const comp = getNativeComponent();
      comp.prop("onIndexChange")(1);
      comp.update();
      expect(comp).toHaveProp(
        "navigationState",
        expect.objectContaining({
          index: 1,
        }),
      );
    });

    it("Check native tab initialLayout ", () => {
      const comp = getNativeComponent();
      expect(comp.prop("initialLayout")).toEqual(initialLayout);
    });

    it("passes right props to renderTabBar", () => {
      const comp = getNativeComponent();
      const expectedProps = {
        globaltheme: "dark",

        0: {key: "first", title: "My Home"},
        1: {key: "second", title: "Rent Passport shares"},

        theme: {
          light: {accentColor: "#5ab88e", background: ["#ffffff", "#ffffff"]},
        },
      };

      comp.prop("renderTabBar")({
        0: {key: "first", title: "My Home"},
        1: {key: "second", title: "Rent Passport shares"},
      });
      expect(TabItem).toHaveBeenCalledWith(
        expect.objectContaining(expectedProps),
      );
    });
  }

  const getNativeComponent = props => {
    const parsedProps = extend(getNativeMockProps(), props);

    return shallow(<Tabs {...parsedProps} />);
  };

  const getWebComponent = props => {
    const parsedProps = extend({}, props);

    return shallow(
      <WebTabs {...parsedProps}>
        <View title="WELCOME">
          <Text>Page 1</Text>
        </View>
        <View title="NATIVE">
          <Text>Page 2</Text>
        </View>
      </WebTabs>,
    );
  };

  const getWebComponentZeroChild = props => {
    const parsedProps = extend({}, props);

    return shallow(<WebTabs {...parsedProps} />);
  };

  const getNativeMockProps = () => ({
    globaltheme: "dark",
    theme: mockTheme,
    scene: {first: page1, second: page2},
    routes: [
      {key: "first", title: "My Home"},
      {key: "second", title: "Rent Passport shares"},
    ],
    initialLayout: {width: Dimensions.get("window").width, height: 47},
  });
});
