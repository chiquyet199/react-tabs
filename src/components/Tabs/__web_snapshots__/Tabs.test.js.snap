// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tabs - page tests matches snapshot 1`] = `
<Styled(View)>
  <Styled(View)
    type="TabsContainer"
  >
    <Connect(WithTheme(Styled(TouchableOpacity)))
      active={true}
      key="WELCOME"
      onPress={[Function]}
    >
      <Connect(WithTheme(Styled(Text)))
        active={true}
      >
        WELCOME
      </Connect(WithTheme(Styled(Text)))>
    </Connect(WithTheme(Styled(TouchableOpacity)))>
    <Connect(WithTheme(Styled(TouchableOpacity)))
      active={false}
      key="NATIVE"
      onPress={[Function]}
    >
      <Connect(WithTheme(Styled(Text)))
        active={false}
      >
        NATIVE
      </Connect(WithTheme(Styled(Text)))>
    </Connect(WithTheme(Styled(TouchableOpacity)))>
  </Styled(View)>
  <Styled(View)
    type="ContentContainer"
  >
    <View
      title="WELCOME"
    >
      <Text>
        Page 1
      </Text>
    </View>
  </Styled(View)>
</Styled(View)>
`;
